### PR TITLE
pre_process data

### PR DESCRIPTION
**Pull request by yukunzGIT**
pre_process data

### CODE REVIEW

File: src/pre_process_nyc_rest.py [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+from sklearn.model_selection import train_test_split
+from sklearn.compose import ColumnTransformer, make_column_transformer
+from sklearn.preprocessing import OneHotEncoder, OrdinalEncoder
+from sklearn.feature_extraction.text import CountVectorizer
+from scipy.sparse import csr_matrix
+
+opt = docopt(__doc__)
+
+def main(input_file, output_train_file, output_test_file):
+
```
Comment (lzung) [inline src/pre_process_nyc_rest.py]: Add comment about importing the raw data

File: src/pre_process_nyc_rest.py [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+import pandas as pd
+from sklearn.model_selection import train_test_split
+from sklearn.compose import ColumnTransformer, make_column_transformer
+from sklearn.preprocessing import OneHotEncoder, OrdinalEncoder
+from sklearn.feature_extraction.text import CountVectorizer
+from scipy.sparse import csr_matrix
+
+opt = docopt(__doc__)
+
+def main(input_file, output_train_file, output_test_file):
```
Comment (lzung) [inline src/pre_process_nyc_rest.py:27]: Include docstring for main

File: src/pre_process_nyc_rest.py [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+    nyc_df = pd.read_csv(input_file)
+    nyc_df.info()
+
+
+    # Clean and modify the data
+    nyc_drop_na_df = nyc_df.dropna()
+    nyc_drop_na_df.info()
+    nyc_drop_na_df.head()
+    nyc_drop_na_df.describe(include='all')
+
```
Comment (lzung) [inline src/pre_process_nyc_rest.py:45]: You should add comments for what you are doing at each step (also outputs like `.info()` and `.head()` and `.describe()` are not needed in the script)

File: src/pre_process_nyc_rest.py [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+    nyc_mod_zipcode_df.loc[nyc_mod_zipcode_df.query("zipcode != @top_20_zipcode").index, 'zipcode'] = 'other_zipcode'
+    
+    nyc_mod_cuisine_df = nyc_mod_zipcode_df.copy()
+    nyc_mod_cuisine_df.loc[nyc_mod_cuisine_df[nyc_mod_cuisine_df['cuisine_description'].map(nyc_mod_cuisine_df['cuisine_description'].value_counts()) < 600].index, 'cuisine_description'] = 'Other_cuisine'
+    
+    nyc_mod_violation_des_df = nyc_mod_cuisine_df.copy()
+    nyc_mod_violation_des_df.loc[nyc_mod_violation_des_df[nyc_mod_violation_des_df['violation_description'].map(nyc_mod_violation_des_df['violation_description'].value_counts()) < 500].index, 'violation_description'] = 'Other_violation_des'
+    
+    nyc_mod_violation_code_df = nyc_mod_violation_des_df.copy()
+    nyc_mod_violation_code_df.loc[nyc_mod_violation_code_df[nyc_mod_violation_code_df['violation_code'].map(nyc_mod_violation_code_df['violation_code'].value_counts()) < 1000].index, 'violation_code'] = 'Other_violation_code'
```
Comment (lzung) [inline src/pre_process_nyc_rest.py]: These modifications should not be done to the data set as a whole before the train/test split. For example, the zipcodes you are using are based on train AND test data, which means the Golden Rule is violated.

File: src/pre_process_nyc_rest.py [hunk 5]
```diff
@@ -0,0 +1,74 @@
+"""Cleans, splits and pre-processes the New York City Restaurant Grading from "https://raw.githubusercontent.com/rfordatascience/tidytuesday/master/data/2018/2018-12-11/nyc_restaurants.csv"
+   Writes the training and test data to separate csv files.
+   
+Usage: pre_process_nyc_rest.py --input_file=<input_file> --output_train_file=<output_train_file> --output_test_file=<output_test_file>
+
+Options:
+--input_file=<input_file>                     Path of the input file from doenload
```
Comment (lzung) [inline src/pre_process_nyc_rest.py]: This should be the "Path of the input file that contains the raw data"

**Review by lzung — CHANGES_REQUESTED** [review]
1) Drop rows that contain NAs in the Grade column.
2) Select rows that are Grade A, B or C ONLY.
3) Transform the target into A vs F (B/C).
4) Train test split
5) Export the data files.

Do not do anything else!!!

**Review by lzung — APPROVED** [review]

**Review by lzung — CHANGES_REQUESTED** [review]
"Each individual script must be well documented. This means comments throughout, readable code and a brief summary at the top of each file that answers who wrote it, when it was written and what it does."

Missing your name and date at the top of the script, comment when you read in the data, comment when you change the label names

Remove the `.info()` and `.head()` line because these are not rendered in the script

Remove the packages that you are not using!

**Review by lzung — APPROVED** [review]
